internal/saucecloud/cloud.go

```diff
@@ -985,12 +987,28 @@ func (r *CloudRunner) getHistory(launchOrder config.LaunchOrder) (insights.JobHi
 	return r.InsightsService.GetHistory(context.Background(), user, launchOrder)
 }
 
+func getSource(isRDC bool) build.Source {
+	if isRDC {
+		return build.RDC
+	}
+	return build.VDC
+}
+
 func (r *CloudRunner) reportSuiteToInsights(res result) {
 	// Skip reporting if job is not completed
 	if !job.Done(res.job.Status) || res.skipped || res.job.ID == "" {
 		return
 	}
 
+	if res.details.BuildID == "" {
+		buildID, err := r.BuildService.GetBuildID(context.Background(), res.job.ID, getSource(res.job.IsRDC))
+		if err != nil {
+			// leave BuildID empty when it failed to get build info
+			log.Warn().Err(err).Str("action", "getBuild").Str("jobID", res.job.ID).Msg(msg.InsightsReportError)
```

Comment:
The error message is wrong now though ;)

Reply:
oh I understand!